channeldb/migration14/migration.go

```diff
@@ -59,112 +65,180 @@ func MigrateDatabaseWireMessages(tx kvdb.RwTx) error {
+		if commitDiffBytes == nil {
+			return nil
+		}
 
-				err = chanBucket.Put(
-					commitDiffKey, b.Bytes(),
-				)
-				if err != nil {
-					return err
-				}
+		// Now that we have the commit diff in the _old_ encoding,
+		// we'll write it back to disk using the new encoding which has
+		// a length prefix in front of the CommitSig.
+		commitDiff, err := deserializeCommitDiffLegacy(
+			bytes.NewReader(commitDiffBytes),
+		)
+		if err != nil {
+			return err
+		}
 
-				// With the commit diff migrated, we'll now
-				// check to see if there're any un-acked
-				// updates we need to migrate as well.
-				updateBytes := chanBucket.Get(
-					unsignedAckedUpdatesKey,
-				)
-				if updateBytes == nil {
-					return nil
-				}
+		var b bytes.Buffer
+		err = serializeCommitDiff(&b, commitDiff)
+		if err != nil {
+			return err
+		}
 
-				// We have un-acked updates we need to migrate
-				// so we'll decode then re-encode them here
-				// using the new format.
-				legacyUnackedUpdates, err := deserializeLogUpdatesLegacy(
-					bytes.NewReader(updateBytes),
-				)
-				if err != nil {
-					return err
-				}
+		err = chanBucket.Put(commitDiffKey, b.Bytes())
+		if err != nil {
+			return err
+		}
 
-				var ub bytes.Buffer
-				err = serializeLogUpdates(
-					&ub, legacyUnackedUpdates,
-				)
-				if err != nil {
-					return err
-				}
+		// With the commit diff migrated, we'll now check to see if
+		// there're any un-acked updates we need to migrate as well.
+		updateBytes := chanBucket.Get(unsignedAckedUpdatesKey)
+		if updateBytes == nil {
+			return nil
+		}
 
-				return chanBucket.Put(
-					unsignedAckedUpdatesKey,
-					ub.Bytes(),
-				)
-			})
+		// We have un-acked updates we need to migrate so we'll decode
+		// then re-encode them here using the new format.
+		legacyUnackedUpdates, err := deserializeLogUpdatesLegacy(
+			bytes.NewReader(updateBytes),
+		)
+		if err != nil {
+			return err
+		}
+
+		var ub bytes.Buffer
+		err = serializeLogUpdates(&ub, legacyUnackedUpdates)
+		if err != nil {
+			return err
+		}
+
+		err = chanBucket.Put(unsignedAckedUpdatesKey, ub.Bytes())
+		if err != nil {
+			return err
+		}
+	}
+
+	// Next, we'll update all the present close channel summaries as well.
+	type closedChan struct {
+		chanKey      []byte
+		summaryBytes []byte
+	}
+	var closedChans []closedChan
+
+	closedChanBucket := tx.ReadWriteBucket(closedChannelBucket)
+	if closedChannelBucket == nil {
+		return fmt.Errorf("no closed channels found")
+	}
+	err = closedChanBucket.ForEach(func(k, v []byte) error {
+		closedChans = append(closedChans, closedChan{
+			chanKey:      k,
+			summaryBytes: v,
 		})
+		return nil
 	})
 	if err != nil {
 		return err
 	}
 
+	for _, closedChan := range closedChans {
+		oldSummary, err := deserializeCloseChannelSummaryLegacy(
+			bytes.NewReader(closedChan.summaryBytes),
+		)
+		if err != nil {
+			return err
+		}
+
+		var newSummaryBytes bytes.Buffer
+		err = serializeChannelCloseSummary(
```

Comment:
> y. But why do the migration of the close summary now then? Why take the risk if we don't need it now and possibly never? 

You brought it up, and it falls under the same case as the other instances fixed in this PR. IMO we might as well bundle these up so we can have one less migration in the future.

Reply:
> For the single-message keys, you could also argue that we should migrate them in case we add something more to it in the future.

Now that we're aware of the issue, a migration isn't the only way to resolve this issue for the single-value use case. They can either use another key or some other option.

Reply:
I brought them up as something to check, but didn't mean to say that they should be migrated if it isn't strictly needed.

Migration isn't the only way to resolve this issue for the single-value use case, but neither is it the only way to resolve the issue for multi-valued use cases. Depending on db structure, another key could be added there too.